### DISCUSSION TITLE
Any feasible way to program schedule?

### DISCUSSION THREAD

**Discussion by dtpanic** [aqualinkd/AqualinkD#207] [General]
I'm assuming this would be a huge update to aqualinkD, but any plans to add support for updating the automation schedule? 

I removed my AqualinkD antenna completely (it was causing noise on the bus) and the only thing I miss is the ability to modify the schedule.

**Comment by sfeakes** [comment]
I looked into this a long time ago, and it wasn’t really worth the time since it’s very difficult to achieve. The panel needs to go into programming mode for a long time and getting all the timings right with unique situations and make it work for everyone’s different setups would take an extremely long time.
But you can clear the programming from the panel all together and use your home automation hub to set up the schedule. You can also use cron on the pi to do this.  At some point I will create a web I to program cron for you.
Example, put the below in crontab, to turn on the pump at 8am and off at 11am every day. 
```
0 8 * * *      curl -X PUT -d value=1  "localhost/api/Filter_Pump/set"
0 11 * * *     curl -X PUT -d value=0  "localhost/api/Filter_Pump/set"
```

**Reply by dtpanic** [reply]
Makes sense thanks. The only thing I don't think I can replicate in a HomeKit schedule is the VSP pump speeds (I run the pump 24x7, at different speed settings).

**Reply by sfeakes** [reply]
Yup there isn’t a great way for HomeKit to support that. But the AqualinkD API does, so you could schedule the speeds through cron like the above example.  You could also use the @boot to turn on the pump when the pi boots (Ie lost power).

**Comment by sfeakes** [comment]
I just posted AqualinkD v2.3.0a, that has scheduling built in.
Read the wiki for instructions.  It uses cron and not the panel for schedules.

**Comment by ballle98** [comment]
> I'm assuming this would be a huge update to aqualinkD, but any plans to add support for updating the automation schedule?
> 
> I removed my AqualinkD antenna completely (it was causing noise on the bus) and the only thing I miss is the ability to modify the schedule.

Another thing that you can do it temporarily stop aqualinkd and use aquaweb as a virtual PDA https://github.com/ballle98/aquaweb then restart aqualinkd
